File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        numberMap = {}

        for i in range(len(nums)):
            diff = target - nums[i]

            if diff in numberMap:
                return [numberMap[diff], i]
            
            numberMap[nums[i]] = i 

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for (int i = 0; i < nums.size(); i++) {
            if (map.find(nums[i]) != map.end()) {
                return vector<int>{map[nums[i]], i};
            }
            map.insert({target - nums[i], i});
        }

        return vector<int>{};
    }
};